### PR TITLE
[kjobctl] Add slurm-init image.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add slurm-init image for kjob.

#### Which issue(s) this PR fixes:

Part of #3243

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/images/slurm-init/Dockerfile
Comment (trasc) [inline cmd/experimental/kjobctl/images/slurm-init/Dockerfile:1]: let's make this something like `cmd/experimental/kjobctl/cmd/slurm-init/Dockerfile`,

I think it's better in the long run to have a go app doing the init setup. (we can make the image smaller)

File: cmd/experimental/kjobctl/Makefile [hunk 1]
```diff
@@ -17,6 +17,14 @@ CGO_ENABLED ?= 0
 GO_CMD ?= go
 GO_FMT ?= gofmt
 
+GIT_TAG ?= $(shell git describe --tags --dirty --always)
+
+DOCKER_BUILDX_CMD ?= docker buildx
+IMAGE_BUILD_CMD ?= $(DOCKER_BUILDX_CMD) build
+STAGING_IMAGE_REGISTRY ?= us-central1-docker.pkg.dev/k8s-staging-images
+IMAGE_REGISTRY ?= $(STAGING_IMAGE_REGISTRY)/kjob
```
Comment (trasc) [inline cmd/experimental/kjobctl/Makefile:25]:
```suggestion
IMAGE_PROJECT ?= kjob
IMAGE_REGISTRY ?= $(STAGING_IMAGE_REGISTRY)/$(IMAGE_PROJECT)
```

This way we can temporary overwrite it in `cloudbuild.yaml` of kueue.

File: cmd/experimental/kjobctl/images/slurm-init/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM alpine:20240807
+
+RUN apk add --no-cache ca-certificates && update-ca-certificates
```
Comment (mwielgus) [inline cmd/experimental/kjobctl/images/slurm-init/Dockerfile:17]: Why do we need these?

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/images/slurm-init/Dockerfile]: You right, it's not need. Thanks!

**Comment by mbobrovskyi** [comment]
/close

Due to kjob moved to https://github.com/kubernetes-sigs/kueue/pull/3269#issuecomment-2516032339.

**Pull request closed without merging**